### PR TITLE
Add machete and remove cargo-udeps

### PR DESCRIPTION
**Pull request by tiruka**
# Related Issues

https://github.com/tracel-ai/xtask/issues/3

# Checks

Looks like test validation process is not established for this repo, I did following checks. If not enough, pleas let me know how to deal with additional checks and tests.

##  No Detection Case

After building, run the command with execution files in the target dir.
As expected no errors occurred.
```
% cargo build                                   
% ./target/xtask/debug/xtask dependencies unused
[2024-09-08T12:23:34Z INFO  tracel_xtask] Execution environment: std
[2024-09-08T12:23:34Z INFO  tracel_xtask::commands::dependencies] Cargo: run unused dependencies checks
[2024-09-08T12:23:34Z INFO  tracel_xtask::utils::process] Command line: cargo machete
Analyzing dependencies of crates in this directory...
cargo-machete didn't find any unused dependencies in .. Good job!
Done!
```

## Detection Case

To detect unused packages, I added `cfg-if` for instances and executed the command.

```
% cargo add cfg-if                    
error: `cargo add` could not determine which package to modify. Use the `--package` option to specify a package. 
available packages: tracel-xtask, tracel-xtask-macros, xtask
toru@toru-mac xtask % cargo add cfg-if --package tracel-xtask
    Updating crates.io index
      Adding cfg-if v1.0.0 to dependencies
             Features:
             - compiler_builtins
             - core
             - rustc-dep-of-std
```

After building, run the command with execution files in the target dir.
As expected an error occurred.
```
% cargo build   
% ./target/xtask/debug/xtask dependencies unused
[2024-09-08T12:24:21Z INFO  tracel_xtask] Execution environment: std
[2024-09-08T12:24:21Z INFO  tracel_xtask::commands::dependencies] Cargo: run unused dependencies checks
[2024-09-08T12:24:21Z INFO  tracel_xtask::utils::process] Command line: cargo machete
Analyzing dependencies of crates in this directory...
cargo-machete found the following unused dependencies in .:
tracel-xtask -- ./crates/tracel-xtask/Cargo.toml:
        cfg-if

If you believe cargo-machete has detected an unused dependency incorrectly,
you can add the dependency to the list of dependencies to ignore in the
`[package.metadata.cargo-machete]` section of the appropriate Cargo.toml.
For example:

[package.metadata.cargo-machete]
ignored = ["prost"]

You can also try running it with the `--with-metadata` flag for better accuracy,
though this may modify your Cargo.lock files.

Done!
Error: Unused dependencies found!
```

### CODE REVIEW

**Comment by tiruka** [comment]
Conflicts is detected for Cargo.lock. It has changed due to my editor (VSCode) automatically, so it may be correct. If downgrade is necessary, I will do that with vim or something else.

File: crates/tracel-xtask/src/commands/dependencies.rs [hunk 1]
```diff
@@ -2,12 +2,8 @@ use anyhow::Ok;
 use strum::IntoEnumIterator;
 
 use crate::{
-    commands::CARGO_NIGHTLY_MSG,
     endgroup, group,
-    utils::{
```
Comment (tiruka) [inline crates/tracel-xtask/src/commands/dependencies.rs:7]: Unnecessary, so delete them.

**Review by Luni-4 — APPROVED** [review]
Thanks a lot for your work! It's fine for me! 

If @syl20bnr agrees, we can merge it

**Review by syl20bnr — APPROVED** [review]
LGTM ! Thank you !

**Pull request merged**